scout/ecm_prep.py

```diff
@@ -1788,9 +1795,256 @@ def __init__(self, base_dir, handyfiles, opts):
         self.env_heat_ls_scrn = (
             "windows solar", "equipment gain", "people gain",
             "other heat gain")
+        # Import/finalize input data on federal or sub-federal incentive level modifications
+        # if state regions are used
+        state_vars = ["incentives", "low_volume_rate"]
+        if opts.alt_regions == "State":
+            self.import_state_data(handyfiles, state_vars, valid_regions, opts)
+        else:
+            for k in state_vars:
+                setattr(self, k, None)
         self.skipped_ecms = []
         self.save_shp_warn = []
 
+    def import_state_data(self, handyfiles, state_vars, valid_regions, opts):
+        """Import and further prepare sub-federal adoption driver data.
+
+        Args:
+            handyfiles (object): File paths.
+            state_vars (list): State-level adoption drivers to read in data for.
+            valid_regions (list): Full list of states run in Scout analyses.
+            opts (object): Stores user-specified execution options.
+        """
+
+        # State-level adoption inputs: modifications to AEO equipment incentive levels
+        for k in state_vars:
+            # Pull the scenario name to use for the current variable; special handling for
+            # the incentives variable, where the user option is split into two sub-options,
+            # incentive_levels, and incentive_restrictions; otherwise, user option name
+            # is the same as variable name
+            if k == "incentives":
+                scn_name = opts.incentive_levels
+            else:
+                scn_name = getattr(opts, k)
+            # Scenario for variable is set to None
+            if not scn_name:
+                setattr(self, k, None)
+                continue
+            # Try importing state-level adoption input data; if not available, set to empty list
+            try:
+                # Read in input-specific data
+                state_econ_dat = pd.read_csv(getattr(handyfiles, k))
+                # Filter by scenarios
+                state_econ_dat = state_econ_dat[state_econ_dat["scenario"] == scn_name]
```

Comment:
This is a really large `try` block and there are a lot of ways in which code could fail other than `FileNotFoundError`. If a file not found error is actually the only reason you want to escape the try block, then I suggest moving the exception up, and moving everything else out of the try/except

```suggestion
            try:
                # Read in input-specific data
                state_econ_dat = pd.read_csv(getattr(handyfiles, k))
             except FileNotFoundError:
                # Set segment-specific list of state-level inputs to empty list
                setattr(self, k, [])
            # Filter by scenarios
            state_econ_dat = state_econ_dat[state_econ_dat["scenario"] == scn_name]
            ...
```

Reply:
Good point. The try/except is in a for loop that runs through each type of adoption input (including incentives and rates), checks for the file, and if it isn't there should move on. So I modified your code to `continue` to the next iteration of that for loop when the file isn't found. Otherwise the code would try to proceed with the rest of the calculations previously under the `try` block (and fail). 

```
           # Try importing state-level adoption input data; if not available, set to empty list
            try:
                # Read in input-specific data
                state_econ_dat = pd.read_csv(getattr(handyfiles, k))
            except FileNotFoundError:
                # Set segment-specific list of state-level inputs to empty list
                setattr(self, k, [])
                # Continue to next adoption input if no data are found
                continue
```